packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -79,6 +79,7 @@ const controlItem = style({
   flexBasis: 0,
   flexGrow: 1,
   flexShrink: 0,
+  minWidth: 0,
```

Comment:
added this to ensure that the text truncates. this enforces that all buttons with track styling have to have the same width. that said, if you don't add a width to the segmented control, it's likely that some text in one of the button's will truncate so you're basically forced to add a width

note: we'll probably need to update this for non-track styling since the width is dependent on the length of the label.

Reply:
that seems fine, you don't actually need a width specified if the parent of the segmented control allocated space via flex or grid

and with track all items are equal size? and without they are variable? if so, then you'll want to change the flexBasis back to undefined for the without track case

Reply:
yeah i've confirmed with design that all items in a track should be equal size and that without a track they are variable. since we only support the track version right now we don't need to worry about without track case but something to keep in mind